src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -947,8 +947,16 @@ private void Open(object parameters)
             // that can't be handled reliably
             try
             {
-                string xmlFilePath = parameters as string;
+                var xmlFilePath = parameters as string;
                 ExecuteCommand(new DynamoModel.OpenFileCommand(xmlFilePath));
+
+                if (UIDispatcher != null)
+                {
+                    // Dispatch a fit view command after all non-idle operations
+                    // are completed. This is to ensure the view's ready for it.
+                    UIDispatcher.BeginInvoke(DispatcherPriority.Background,
+                        (Action)(() => { if (CanFitView(null)) FitView(null); }));
+                }
             }
```

Comment:
Hi @pboyer, it's understandable that it raise a few questions, so here's what I'm thinking: this is done as part of the `Open` command, and that's on the UI thread. Naturally I first tried to do so with just a fit view method call, and nothing happened. The reason for that is this open action creates a bunch `ModelBase` objects in the model layer, and it won't be until a much later time that it gets reflected on the view layer. Doing a fit view before view gets generated results in a no-op. That's why I introduced a `BeginInvoke` call.

But that alone wasn't enough, if for some reason the `BeginInvoke` happens before UI layout is completely updated, then the fit view will not happen too. In that case the fit view occasionally happens, but not always. So I reduced the priority to `DispatcherPriority.Background`, that's when it consistently get executed as expected. Does that make sense?